### PR TITLE
feat: add Ingress for opencost UI

### LINKED ISSUE #32: Add Ingress for UI
Just a basic feature.

### PR DESCRIPTION
**Pull request by boniek83**
Fixes #32

### CODE REVIEW

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
Hi @boniek83 

I am only a contributor here and not a maintainer so my review is not that relevant ;-)

But I'd add an Ingress template which is almost identical with the `helm create mychart` template/skeleton.
The reason behind this is that OpenCost is per definition supported from Kubernetes versions >1.8:

> **What Kubernetes versions does OpenCost support?**
> OpenCost supports a Kubernetes environment above 1.8.
>
> Source: https://www.opencost.io/docs/FAQ#what-kubernetes-versions-does-opencost-support

Therefore the skeleton would be perfect backward compatible:

```yaml
{{- if .Values.ingress.enabled -}}
{{- $fullName := include "foobar.fullname" . -}}
{{- $svcPort := .Values.service.port -}}
{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
  {{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
  {{- end }}
{{- end }}
{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: networking.k8s.io/v1
{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: networking.k8s.io/v1beta1
{{- else -}}
apiVersion: extensions/v1beta1
{{- end }}
kind: Ingress
metadata:
# ...
```

**Comment by mattray** [comment]
@boniek83 do you want to merge in @mkilchhofer's suggestion? We'll merge afterwards

**Comment by mkilchhofer** [comment]
> @boniek83 do you want to merge in @mkilchhofer's suggestion? We'll merge afterwards

@mattray where does this statement comes from. Do you still support kubernetes 1.8 as of 2023? ;) 
As the chart does not follow this today.

In kubernetes 1.8, the following APIs are not available:
- Deployment: `apps/v1` is not available (needs to be `apps/v1beta1`)
- ClusterRoles / ClusterRoleBindings: `rbac.authorization.k8s.io/v1` not available (needs to be `rbac.authorization.k8s.io/v1beta1`)
- ...

**Comment by boniek83** [comment]
@mattray I can do that, but you don't support kubernetes 1.8 anywhere else in this helm chart, so I wonder if there is a point to that. Things can always be changed later if issue of compatibility pops up, which is not likely IMHO. Additionally is it worth it to support such old versions? Helm chart will get increasingly complicated as it gains more features and as kubernetes versions keep getting bigger.

**Review by mattray — APPROVED** [review]

**Comment by mattray** [comment]
Sorry, I blanked on that thinking of another recent breaking change and my knee-jerk response was "backwards compatibility good" ;)

**Comment by boniek83** [comment]
I will fix conflict tomorrow.

**Comment by boniek83** [comment]
@mattray  Maintainers will have to bump helm chart version or should I do it?

**Comment by mattray** [comment]
@boniek83 you can do it or we (the maintainers) will do it manually after merging. It would make sense for us to automate that too.

**Comment by mkilchhofer** [comment]
> @boniek83 you can do it or we (the maintainers) will do it manually after merging. It would make sense for us to automate that too.

🤔 I automated it in:
- #35
- #30

Contributors should bump it IMHO. This is the standard workflow for chart contributions in the helm community.

File: charts/opencost/values.yaml [hunk 1]
```diff
@@ -136,6 +136,15 @@ opencost:
...
 
+    # -- Ingress for OpenCost UI
+    ingress:
+      enabled: false
+      ingressClassName: ""
+      hosts:
+        - host: example.local
+          paths:
+            - /
+
```
Comment (mkilchhofer) [inline charts/opencost/values.yaml:147]: Please document all possible values separately:

```suggestion
    ingress:
      # -- Ingress for OpenCost UI
      enabled: false
      # -- Ingress controller which implements the resource
      ingressClassName: ""
      # -- Annotations for Ingress resource
      annotations: {}
        # kubernetes.io/tls-acme: "true"
      # -- A list of host rules used to configure the Ingress
      # @default -- See [values.yaml](values.yaml)
      hosts:
        - host: example.local
          paths:
            - /
      # -- Ingress TLS configuration
      tls: []
        #  - secretName: chart-example-tls
        #    hosts:
        #      - chart-example.local
```

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
After implementing my proposed changes, please run helm-docs again.

**Pull request merged**